R/derive_extreme_event.R

```diff
@@ -486,6 +482,248 @@ derive_extreme_event <- function(dataset,
+#'       set_values_to = exprs(LSTALVDT = ADT, DTHFL = "N")
+#'     ),
+#'     event(
+#'       dataset_name = "adsl",
+#'       condition = !is.na(TRTEDT),
+#'       order = exprs(TRTEDT),
+#'       mode = "last",
+#'       set_values_to = exprs(LSTALVDT = TRTEDT, DTHFL = "N")
+#'     )
+#'   ),
+#'   source_datasets = list(adsl = adsl, lb = lb),
+#'   order = exprs(USUBJID, LSTALVDT, event_nr),
+#'   mode = "last",
+#'   keep_source_vars = exprs(STUDYID, USUBJID),
+#'   new_vars = exprs(LSTALVDT = LSTALVDT, DTHFL = DTHFL)
+#' )
+#'
+#' # Derive DTHCAUS from AE and DS domain data
+#' adsl <- tribble(
+#'   ~STUDYID,  ~USUBJID,
+#'   "STUDY01", "PAT01",
+#'   "STUDY01", "PAT02",
+#'   "STUDY01", "PAT03"
+#' )
+#' ae <- tribble(
+#'   ~STUDYID,  ~USUBJID, ~AESEQ, ~AEDECOD,       ~AEOUT,  ~AEDTHDTC,
+#'   "STUDY01", "PAT01",  12,     "SUDDEN DEATH", "FATAL", "2021-04-04",
+#'   "STUDY01", "PAT01",  13,     "CARDIAC ARREST", "FATAL", "2021-04-03",
+#' )
+#'
+#' ds <- tribble(
+#'   ~STUDYID, ~USUBJID, ~DSSEQ, ~DSDECOD, ~DSTERM, ~DSSTDTC,
+#'   "STUDY01", "PAT02", 1, "INFORMED CONSENT OBTAINED", "INFORMED CONSENT OBTAINED", "2021-04-03",
+#'   "STUDY01", "PAT02", 2, "RANDOMIZATION", "RANDOMIZATION", "2021-04-11",
+#'   "STUDY01", "PAT02", 3, "DEATH", "DEATH DUE TO PROGRESSION OF DISEASE", "2022-02-01",
+#'   "STUDY01", "PAT03", 1, "DEATH", "POST STUDY REPORTING OF DEATH", "2022-03-03"
+#' )
+#'
+#' derive_vars_extreme_event(
+#'   adsl,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   events = list(
+#'     event(
+#'       dataset_name = "ae",
+#'       condition = AEOUT == "FATAL",
+#'       set_values_to = exprs(DTHCAUS = AEDECOD, DTHDT = convert_dtc_to_dt(AEDTHDTC)),
+#'       order = exprs(DTHDT)
+#'     ),
+#'     event(
+#'       dataset_name = "ds",
+#'       condition = DSDECOD == "DEATH" & grepl("DEATH DUE TO", DSTERM),
+#'       set_values_to = exprs(DTHCAUS = DSTERM, DTHDT = convert_dtc_to_dt(DSSTDTC)),
+#'       order = exprs(DTHDT)
+#'     )
+#'   ),
+#'   source_datasets = list(ae = ae, ds = ds),
+#'   tmp_event_nr_var = event_nr,
+#'   order = exprs(USUBJID, DTHDT, event_nr),
+#'   mode = "first",
+#'   keep_source_vars = exprs(STUDYID, USUBJID),
+#'   new_vars = exprs(DTHCAUS = DTHCAUS, DTHDT = DTHDT)
+#' )
+derive_vars_extreme_event <- function(dataset,
+                                      by_vars = NULL,
+                                      events,
+                                      tmp_event_nr_var = NULL,
+                                      order,
+                                      mode,
+                                      source_datasets = NULL,
+                                      check_type = "warning",
+                                      new_vars,
+                                      keep_source_vars = by_vars) {
+  assert_expr_list(new_vars)
+
+  new_obs <- derive_extreme_event(
+    dataset = NULL,
+    by_vars = by_vars,
+    events = events,
+    tmp_event_nr_var = event_nr,
```

Comment:
Should be `!!tmp_event_nr_var` instead of `event_nr`.

Reply:
Updated, thanks!